### ISSUE TITLE
IG error in TF 2.3.0

### ISSUE THREAD

**Issue by Realvincentyuan** [SeldonIO/alibi#285]
I met this error below when running TF 2.3.0 in order to apply the IG on the tabular data. I did not find the example of apply IG on tabular data so I plan to give it a shot.

## Code
```
n_steps = 50
method = "gausslegendre"
internal_batch_size = 100
nb_samples = 10
ig  = IntegratedGradients(model,
                          layer=model.layers[1],
                          n_steps=n_steps,
                          method=method,
                          internal_batch_size=internal_batch_size)
```

## Error
```
AttributeError: 'Sequential' object has no attribute '_nested_inputs'
```

**Comment by gipster** [comment]
Thanks for flagging the issue. 
It looks like something related to the model your are feeding to the IG class. Is your model defined using tf.keras layers or keras layers?

**Comment by Realvincentyuan** [comment]
> Thanks for flagging the issue.
> It looks like something related to the model your are feeding to the IG class. Is your model defined using tf.keras layers or keras layers?

Thanks for replying! I am using tf.keras layers. Do you have plans to release the official version of IG on tabular data?

**Comment by gipster** [comment]
There are no examples at the moment, but the present version It should work on tabular data as well. 
I tried to reproduce the error but I wasn't able to. Can you please share the code defining your model or some minimal code that produce this error. Thanks

**Comment by Realvincentyuan** [comment]
> There are no examples at the moment, but the present version should work on tabular data as well.
> I tried to reproduce the error but I wasn't able to. Can you please share the code defining your model or some minimal code that produce this error. Thanks

yes, I agree that the current release would work for tabular data.

I am using this sample data and sample code as shown in the [TF official doc](https://www.tensorflow.org/tutorials/structured_data/feature_columns). Please take a look.

**Comment by gipster** [comment]
I managed to reproduce the error. The error originates from the input layer of the model  in the [TF docs example](https://www.tensorflow.org/tutorials/structured_data/feature_columns) you linked. At the moment IG only supports models taking numpy array inputs and having standard keras or tf.keras input layers. 

We are investigating whether it is possible to support other types of inputs, I'll create a new dedicated issue.

**Comment by Realvincentyuan** [comment]
> I managed to reproduce the error. The error originates from the input layer of the model in the [TF docs example](https://www.tensorflow.org/tutorials/structured_data/feature_columns) you linked. At the moment IG only supports models taking numpy array inputs and having standard keras or tf.keras input layers.
> 
> We are investigating whether it is possible to support other types of inputs, I'll create a new dedicated issue.

Thank you for the quick turnaround. Please pin me in the update of this issue. I believe this update is going to help a lot of people!

**Comment by Realvincentyuan** [comment]
@gipster Hi, do you have update?

**Comment by jklaise** [comment]
Hi @Realvincentyuan , we are currently not actively investigating supporting models that take data other than arrays as the number of use cases is limited (I'm actually surprised that the official Tensorflow example shows such a non-standard workflow) but the changes required to Alibi would be quite significant. We do want to support this in the future but for the time being I would recommend writing your tabular data Tensorflow models in a way such that the input pipeline begins with arrays.

**Comment by Realvincentyuan** [comment]
> Hi @Realvincentyuan , we are currently not actively investigating supporting models that take data other than arrays as the number of use cases is limited (I'm actually surprised that the official Tensorflow example shows such a non-standard workflow) but the changes required to Alibi would be quite significant. We do want to support this in the future but for the time being I would recommend writing your tabular data Tensorflow models in a way such that the input pipeline begins with arrays.

Yes, makes sense, looks like there is no example of applying IG on tabular data so far. I will see if I can get inspiration from your solution on Text. Thanks.

**Comment by NMRobert** [comment]
Hi - I feel that the linked data input workflow is now becoming the standard one, and it would be terrific to have Alibi support for it.
